andela/annie-hall-cfh
# Chore(npm to yarn): Use yarn as package manager instead of npm, and u…

###What does this PR do?

Use yarn as package manager instead of npm, and upgrade packages

###Description of Task to be completed?

* Install yarn package manager
* Update each npm package in package.json file using yarn
* Manually install express 4 independent npm packages
* Modify `engine` property of package.json file to latest node and npm versions
* Modify express.js file to reflect proper usage of `express 4` independent npm packages
* Import apiRoute path from `routes.js` file into `express.js` file and use appropriately
* Initialize `express router` in `routes.js` file, and change 'app' to parameter to 'router' variable. Then export the `router` variable
* Find and fix upgrade errors in 404.jade, 500.jade and default.jade files

###How should this be manually tested?

####The following steps would run the application

 1. From your terminal or console window, run `brew install yarn` 
 2. Complete the basic setup of the project gotten from the README, on your local repository.
 3. Cd to the root of the project folder in your console. 
 4. First run the command "yarn install" to install all required and updated node packages including 
     bower components
 5. Run the command `yarn global add gulp-cli`
 6. Create a .env file in root of project folder and add 
     `MONGOHQ_URL=mongodb://localhost:27017/cfh` inside the file
 7. Make sure you have downloaded Mongo DB  and set it up on your PC. Use this => [Download and Setup MongoDB](https://docs.mongodb.com/manual/administration/install-community/)
 8. Create and setup a database using this file [Setup Database](https://docs.mongodb.com/manual/mongo/). Make sure you name the database as `cfh` 
 9. Once `mongod` server is running, run the command `yarn start` or `gulp` to start the application.
 10. Finally, navigate to your browser and enter `localhost:3000`

Any background context you want to provide?

Nil

What are the relevant pivotal tracker stories?

`#152798631`

Screenshots

<img width="733" alt="screen shot 2017-11-29 at 6 21 39 pm" src="https://user-images.githubusercontent.com/22524619/33389205-5fc3631a-d532-11e7-8f88-208595d85b55.png">


<img width="1184" alt="screen shot 2017-11-27 at 2 05 22 pm" src="https://user-images.githubusercontent.com/22524619/33268133-590f0cca-d37c-11e7-84ca-c3eac9b9da03.png">

<img width="1433" alt="screen shot 2017-11-27 at 2 06 37 pm" src="https://user-images.githubusercontent.com/22524619/33268145-64bd3ccc-d37c-11e7-9b40-ace26d9bf48b.png">
